src/Workspaces/Core/Portable/Shared/Extensions/DocumentExtensions.cs

```diff
@@ -59,6 +59,11 @@ public static bool IsOpen(this Document document)
                 }
 
                 var node = token.Parent.AncestorsAndSelf().FirstOrDefault(a => a.FullSpan.Contains(span));
+                if (node == null)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(span));
```

Comment:
Seems more reasonable, should we just do if (!root.FullSpan.Contains(span)) earlier and throw instead? This seems to be a fairly circuitous way to check the simpler case.